### PR TITLE
EWS should email PR author in case Merge-Queue silently fails on their PR

### PR DESCRIPTION
**Pull request by aj062**
#### 75f724fe64e48eae8115575c2ce05b371a75218b
<pre>
EWS should email PR author in case Merge-Queue silently fails on their PR
<a href="https://bugs.webkit.org/show_bug.cgi?id=241101">https://bugs.webkit.org/show_bug.cgi?id=241101</a>

Reviewed by Jonathan Bedard.

* Tools/CISupport/ews-build/steps.py:
(ValidateChange.validate_github):
(ValidateChange.send_email_for_github_failure):

Canonical link: <a href="https://commits.webkit.org/251148@main">https://commits.webkit.org/251148@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@295053">https://svn.webkit.org/repository/webkit/trunk@295053</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

**Comment by aj062** [comment]
Sample run: https://ews-build.webkit-uat.org/#/builders/85/builds/242

File: Tools/CISupport/ews-build/steps.py [hunk 1]
```diff
@@ -1568,22 +1568,59 @@ def validate_github(self, pr_number):
...
-            self.skip_build("PR {} does not have a merge queue label".format(pr_number))
-            return False
+        if self.verifyMergeQueue:
+            if not pr_json:
+                self.send_email_for_github_failure()
+                self.skip_build("Infrastructure issue: unable to check PR status")
+                return False
+            merge_queue = self._is_pr_in_merge_queue(pr_json)
+            if merge_queue == 0:
+                self.skip_build("PR {} does not have a merge queue label".format(pr_number))
```
Comment (JonWBedard) [inline Tools/CISupport/ews-build/steps.py:1578]: I think sending an email in this case is a good idea, but I don't know that I would classify this case as an "infrastructure failure"

**Review by JonWBedard — APPROVED** [review]

File: Tools/CISupport/ews-build/steps.py (hunk 1 above)
Comment (aj062) [reply Tools/CISupport/ews-build/steps.py:1578]: Note that we are only classifying it as infrastructure issue when we are actually unable to fetch pr json from github, which is an infrastructure issue from engineer's perspective. 
We aren't classifying PR not having merge queue label as infra issue.

Comment (JonWBedard) [reply Tools/CISupport/ews-build/steps.py:1578]: Line 1612:
```
email_subject = f'Infrastructure failure on {builder_name} for PR #{pr_number}: {title}'
```

Comment (JonWBedard) [reply Tools/CISupport/ews-build/steps.py:1578]: Wait, we don't send an email here, missed that.

**Review by JonWBedard — APPROVED** [review]

**Pull request merged**